parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -766,38 +825,43 @@ pub mod pallet {
 			message: H256,
 			signature: BoundedVec<u8, T::MaxSignatureSize>,
 		) -> DispatchResult {
-			let sig = Signature::from_bytes(&signature[..]);
-			if let Err(_e) = sig {
-				return Err(Error::<T>::InvalidSignature.into())
-			}
+			let agg_sig = Self::bls_aggregate_signature(signature)?;
```

Comment:
I don't think this refactoring is an improvement. The old version was easier to understand as all the logic was in a single function, and the function wasn't too large. This helps auditors.

Refactoring the method would make sense if `bls_aggregate_signature`, etc, was going to be used in multiple places, but thats not the case here.

Substrate's benchmarking framework allows to benchmark arbitrary code in `{ ... }`, and so I would rather use that approach.